mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/model/path/PojoModelPathPropertyNode.java

```diff
@@ -25,8 +26,17 @@ public final class PojoModelPathPropertyNode extends PojoModelPath {
 	private final String propertyName;
 
 	PojoModelPathPropertyNode(PojoModelPathValueNode parent, String propertyName) {
-		this.parent = parent;
 		Contracts.assertNotNullNorEmpty( propertyName, "propertyName" );
+		Contracts.assertTrue(
+				DOT_PATTERN.split( propertyName, -1 ).length == 1,
+				String.format(
+						Locale.ROOT,
+						"Property name '%s' is invalid. The value must match a property in a corresponding class.",
+						propertyName
+				)
```

Comment:
You should probably add a new `@Message` method that returns String to the appropriate `Log` class. Otherwise this message cannot be localized, and given it is rather specific to Hibernate Search, I'd rather leave the door open to localization.

Personally, I would even skip the `assertTrue` thingy and just add a method that returns an exception to the `Log` class, but it's up to you.

Also, about the message... can't we simply say `property names must not contain any dot (".")`? Since, well, the purpose of this issue was to produce clearer error messages :)
We can always elaborate as to what property names represent, like you did, but I'd only do that after explaining what's wrong exactly.

Reply:
>Also, about the message... 

I was torn about this part `or any metacharacter` even though the pattern seems to be using just the dot. So I wasn't sure if the set of characters will change and it won't be just about the dots or if it'll remain as is. anyway - the message is updated 😃